common/include/scipp/common/deep_ptr.h

```diff
@@ -13,13 +14,25 @@ template <class T> class deep_ptr {
   using element_type = typename std::unique_ptr<T>::element_type;
   deep_ptr() = default;
   deep_ptr(std::unique_ptr<T> &&other) : m_data(std::move(other)) {}
-  deep_ptr(const deep_ptr<T> &other)
-      : m_data(other ? std::make_unique<T>(*other) : nullptr) {}
+  deep_ptr(const deep_ptr<T> &other) {
+    if (other) {
+      if constexpr (std::is_abstract<T>())
+        *this = other->clone();
```

Comment:
This is the fix for some kind of copying problem with unique_ptr abstract type? Do you recall how that came about from the switch to shared libraries?

Reply:
It is necessary for Windows. Not sure what is going on. The methods are hidden for abstract types, but somehow it is trying to instantiate this, probably because it is trying to create the exported symbols, despite being hidden?